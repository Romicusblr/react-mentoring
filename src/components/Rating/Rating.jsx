import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './Rating.module.css';

function Rating({ children, className, ...other }) {
  return (
    <span
      className={classNames(style.rating, className)}
      {...other}
    >
      {children}
    </span>
  );
}

Rating.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Rating.defaultProps = {
  className: '',
};

export default Rating;
