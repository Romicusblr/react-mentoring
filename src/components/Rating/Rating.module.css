.rating {
    --rating-size: 4rem;
    width: var(--rating-size);
    height: var(--rating-size);
    display: flex;
    align-items: center;
    justify-content: center;
    color: var(--color-rating);
    border: 1px solid var(--color-white);
    border-radius: 50%;
}
