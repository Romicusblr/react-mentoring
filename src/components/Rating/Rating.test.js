import Rating from '.';

describe('Rating', () => {
  it('should match to snapshot', () => {
    const snap = shallow(<Rating>7.0</Rating>);
    expect(snap).toMatchSnapshot();
  });
});
