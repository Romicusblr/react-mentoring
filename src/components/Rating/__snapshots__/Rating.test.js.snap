// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should match to snapshot 1`] = `
<span
  className="rating"
>
  7.0
</span>
`;
